model-optimizer/mo/front/mxnet/extractors/utils.py

```diff
@@ -101,21 +101,25 @@ def has(self, key):
 
 def get_mxnet_node_edges(node: dict, node_id: [int, str], nodes_list: list, index_node_key: dict):
     edge_list = []
+    used_indices = []
     for in_port, src_node_id in enumerate(node['inputs']):
-        src_node = src_node_id[0]
-        dest_port = src_node_id[1]
-        edge_attrs = {
-            'in': in_port,
-            'out': dest_port,
-            # debug anchor for framework name, out port and tensor name
-            'fw_tensor_debug_info': [(index_node_key[src_node], src_node_id[1], nodes_list[src_node]['name'])],
-            'in_attrs': ['in'],
-            'out_attrs': ['out'],
-            'data_attrs': ['fw_tensor_debug_info']
-        }
-        edge = (index_node_key[src_node], index_node_key[node_id], edge_attrs)
+        edge = create_mxnet_edge(index_node_key[src_node_id[0]], index_node_key[node_id], in_port, src_node_id[1],  nodes_list[src_node_id[0]]['name'])
         edge_list.append(edge)
-    return edge_list
+        used_indices.append(src_node_id[0])
+    return edge_list, used_indices
+
+
+def create_mxnet_edge(src_node_id: str, dst_node_id: str, src_port: int, dst_port: int, framework_name: str):
+    edge_attrs = {
+        'in': src_port,
+        'out': dst_port,
+        # debug anchor for framework name, out port and tensor name
+        'fw_tensor_debug_info': [(framework_name, dst_port, framework_name)],
```

Comment:
In caffe 2 names used: bottom and saved layer_name. Here only one. Why? Also add parameter just to return it looks strange for me.

Reply:
'fw_tensor_debug_info' should contain (framework operation name, destination port, tensor name). In caffe tensor names are stored in "bottom" attribute and there is also an operation name. But in mxnet there are no tensor names, only operation names. So the operation name is used twice.